geth/jail/cell.go

```diff
@@ -15,9 +15,64 @@ import (
 
 const timeout = 5 * time.Second
 
+// JailManager defines methods for managing jailed environments
+type JailManager interface {
+	// Call executes given JavaScript function w/i a jail cell context identified by the chatID.
+	Call(chatID, this, args string) string
+
+	// CreateCell creates a new jail cell.
+	CreateCell(chatID string) (JailCell, error)
+
+	// Parse creates a new jail cell context, with the given chatID as identifier.
+	// New context executes provided JavaScript code, right after the initialization.
+	// DEPRECATED in favour of CreateAndInitCell.
+	Parse(chatID, js string) string
+
+	// CreateAndInitCell creates a new jail cell and initialize it
+	// with web3 and other handlers.
+	CreateAndInitCell(chatID string, code ...string) string
+
+	// Cell returns an existing instance of JailCell.
+	Cell(chatID string) (JailCell, error)
+
+	// Execute allows to run arbitrary JS code within a cell.
+	Execute(chatID, code string) string
+
+	// SetBaseJS allows to setup initial JavaScript to be loaded on each jail.CreateAndInitCell().
+	SetBaseJS(js string)
+
+	// Stop stops all background activity of jail
+	Stop()
+}
+
+// JSValue is a wrapper around an otto.Value.
+type JSValue struct {
+	value otto.Value
+}
+
+// Value returns the underlying otto.Value from a JSValue. This value IS NOT THREADSAFE.
+func (v *JSValue) Value() otto.Value {
```

Comment:
I think we can even make it non-exportable, because it seems to be used only for test.

Reply:
Will do!

Reply:
Yeah, then we can leave it as is, I think